### PR TITLE
Fix refs inconsistency on hot reloading

### PR DESCRIPTION
**Pull request by umidbekkarimov**
Related to #18

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -101,4 +101,85 @@ describe('react-jss', () => {
       expect(isRendered).to.be(true)
     })
   })
+
+  describe('.injectSheet() hot reloading', () => {
+    function imitateHotReloading(container, TargetClass, SourceClass) {
```
Comment (kof) [inline src/index.test.js]: maybe simulateHotReloading is a better name

File: src/index.js [hunk 2]
```diff
@@ -62,8 +62,12 @@ function wrap(jss, WrappedComponent, styles, options = {}) {
     }
 
     componentWillUnmount() {
-      deref()
-      this.sheet = null
+      if (this.sheet && !sheet && !refs) {
+        this.sheet.detach()
+      }
```
Comment (kof) [inline src/index.js]: you can just do this:

} deref()

**Comment by kof** [comment]
2 small styling comments, otherwise fine.

File: src/index.test.js (hunk 1 above)
Comment (kof) [inline src/index.test.js]: then lets keep it that way for consistency.

Comment (kof) [inline src/index.test.js]: oh you renamed it, its ok!

**Comment by kof** [comment]
merged!

**Comment by umidbekkarimov** [comment]
@kof when this pr can be published?

**Comment by kof** [comment]
published

**Pull request merged**